lib/Yield/CrossRelations.cpp

```diff
@@ -143,19 +155,23 @@ yield::Graph yield::CrossRelations::toYieldGraph() const {
   };
   auto AddEdge = [&LookupHelper](std::string_view FromLocation,
                                  std::string_view ToLocation,
-                                 yield::RelationType::Values) {
-    auto FromKey = p::genericLocationFromString<0>(FromLocation,
+                                 yield::RelationType::Values Kind) {
+    if (Kind == yield::RelationType::IsCalledFrom) {
+      auto FromKey = p::genericLocationFromString<0>(FromLocation,
+                                                     ranks::Function,
+                                                     ranks::BasicBlock,
+                                                     ranks::Instruction);
+      auto ToKey = p::genericLocationFromString<0>(ToLocation,
                                                    ranks::Function,
                                                    ranks::BasicBlock,
                                                    ranks::Instruction);
-    auto ToKey = p::genericLocationFromString<0>(ToLocation,
-                                                 ranks::Function,
-                                                 ranks::BasicBlock,
-                                                 ranks::Instruction);
-    revng_assert(FromKey.has_value() && ToKey.has_value());
-
-    auto *FromNode = LookupHelper.at(std::get<0>(FromKey.value()));
-    FromNode->addSuccessor(LookupHelper.at(std::get<0>(ToKey.value())));
+      revng_assert(FromKey.has_value() && ToKey.has_value());
+
+      auto *FromNode = LookupHelper.at(std::get<0>(FromKey.value()));
+      auto *ToNode = LookupHelper.at(std::get<0>(ToKey.value()));
+      if (!llvm::is_contained(FromNode->successors(), ToNode))
+        FromNode->addSuccessor(ToNode);
+    }
```

Comment:
Very vague commit message

Reply:
Improved message with better description and split it into two different commits.